### PR TITLE
sdl2-ttf: Add version 2.0.18

### PR DESCRIPTION
**Pull request by criticic**
Added the sdl2_ttf library, which is used for font rendering.

Relates to #7867
Thanks to @g-berthiaume as this is based on sdl2-image manifest.

- [x] I have read the [Contributing Guide](https://github.com/ScoopInstaller/.github/blob/main/.github/CONTRIBUTING.md).

### CODE REVIEW

File: bucket/sdl2-ttf.json [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+            "if($was_in_path) { $env:CMAKE_PREFIX_PATH = $newpath }"
+        ]
+    },
+    "checkver": {
+        "url": "https://github.com/libsdl-org/SDL_ttf/releases/download/latest/",
+        "regex": "SDL2_ttf-([\\d.]+)\\.zip"
+    },
+    "autoupdate": {
+        "url": [
+            "https://github.com/libsdl-org/SDL_ttf/releases/download/release-2.0.18/SDL2_ttf-devel-$version-VC.zip",
```
Comment (versedwildcat) [inline bucket/sdl2-ttf.json:125]: Missed a $version for the update URLs
`https://github.com/libsdl-org/SDL_ttf/releases/download/release-$version/SDL2_ttf-devel-$version-VC.zip"`

**Comment by issaclin32** [comment]
The manifest works fine for me, but I think this should be in [the main bucket](https://github.com/ScoopInstaller/Main).

Also, please use `checkver`->`regex` instead of `re`. 
`re` is now only kept for backward-compatibility.

**Comment by rashil2000** [comment]
The manifest doesn't follow the 4th [criterion](https://github.com/ScoopInstaller/Scoop/wiki/Criteria-for-including-apps-in-the-main-bucket) for Main bucket.

**Comment by criticic** [comment]
/verify

> Also, please use checkver->regex instead of re.
> re is now only kept for backward-compatibility.
Few examples here still use re, https://github.com/ScoopInstaller/Scoop/wiki/App-Manifest-Autoupdate, dont know from where it sourced though.

**Comment by issaclin32** [comment]
> Few examples here still use re, https://github.com/ScoopInstaller/Scoop/wiki/App-Manifest-Autoupdate, dont know from where it sourced though.

Thank you for reporting. I have updated the page.

**Comment by issaclin32** [comment]
> The manifest doesn't follow the 4th [criterion](https://github.com/ScoopInstaller/Scoop/wiki/Criteria-for-including-apps-in-the-main-bucket) for Main bucket.

I see.

**Pull request merged**